### PR TITLE
MOTECH-2731: Migrate Module Settings

### PR DESCRIPTION
**Pull request by mkwiatkowskisoldevelo**
- added module settings in admin
- added file-upload directive
- settings for task module still won't work, because task settings page isn't implemented yet

### CODE REVIEW

File: src/admin/bundles/bundle-settings.controller.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            } else {
+                successHandler = function () {
+                    LoadingModal.close();
+                    ModalFactory.showSuccessAlert('admin.settings.saved');
+                };
+            }
+
+            LoadingModal.open();
+
+            $(id).ajaxSubmit({
```
Comment (jredlarski) [inline src/admin/bundles/bundle-settings.controller.js]: As suggested in [PR#39](https://github.com/motech/motech-ui/pull/39) we should not use JQueryFormPlugin as it is rather unsupported.

File: src/admin/bundles/bundle-settings.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            <input type="submit" ng-click="saveSettings(mSettings)" value="{{'admin.submit' | translate}}"/>
+            <input type="submit" ng-click="saveSettings(mSettings, true)" value="{{'admin.submitAndRestart' | translate}}"/>
+        </form>
+    </div>
+
+    <div class="settings-form" ng-repeat="rawFile in rawFiles">
+        <h4 class="settings-section">{{rawFile}}</h4>
+        <form id="_raw_{{rawFile}}" name="{{rawFile}}Form">
+            <div class="form-group fileinput fileinput-new" data-provides="fileinput">
+                <label>{{msg('admin.button.selectFile')}}</label>
```
Comment (jredlarski) [inline src/admin/bundles/bundle-settings.html]: Use translate

File: src/common/file-upload/messages/file-upload.en.properties [hunk 3]
```diff
@@ -0,0 +1 @@
+fileInput.select="Select"
```
Comment (jredlarski) [inline src/common/file-upload/messages/file-upload.en.properties]: Do we need this one? I can't rely see a place where it is used...

File: src/common/file-upload/file-upload.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    <span class="fileinput-preview"></span>
+</div>
+<span class="btn btn-default btn-file">
+        <span class="fileinput-new">{{msg('server.button.selectFile')}}</span>
+        <span class="fileinput-exists">{{msg('server.button.changeFile')}}</span>
+        <input type="file" id="fileInput" name="file" class="input-file"/>
+    </span>
+<a href="#" class="btn btn-danger fileinput-exists" data-dismiss="fileinput">
+    <i class="fa fa-fw fa-trash-o"></i>&nbsp;{{msg('server.button.remove')}}
+</a>
```
Comment (jredlarski) [inline src/common/file-upload/file-upload.html]: Couldn't it be a `<button></button>` element with a class `button-delete` from [common/base](https://github.com/motech/motech-ui/tree/master/src/common/base)?

Comment (jredlarski) [inline src/common/file-upload/file-upload.html]: Add translate to the translatable strings

**Comment by jredlarski** [comment]
@mkwiatkowskisoldevelo Here are some issues that I have found in your PR:

**(1)** We shouldn't use JQueryFormPlugin so lets drop that.
**(2)** Please add translate to the translatable stings and refactor html a bit. I've made some inline comments for that.
**(3)** You have added an empty 'file-upload.css' file and 'file-upload.en.properties' file with a variable which isn't used. I believe both should be removed, right?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: src/common/file-upload/file-upload.html (hunk 4 above)
Comment (shanmukhm) [inline src/common/file-upload/file-upload.html]: @jredlarski I'm already migrating motech-file-upload. Would it become a duplicate?

File: src/admin/bundles/bundle-settings.html [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                <div class="col-sm-9 col-md-10">
```
Comment (jredlarski) [inline src/admin/bundles/bundle-settings.html]: Try not to use the `col-` classes

**Comment by jredlarski** [comment]
@mkwiatkowskisoldevelo I've added a motech-file-upload directive in a separate [PR](https://github.com/motech/motech-ui/pull/46) so please use it and resolve any conflicts if they occur.

File: src/admin/nav.html [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <ul>
     <li ui-sref-active="active"><a ui-sref="bundles">{{ 'admin.manageModules' | translate }}</a></li>
     <li><a>{{ 'admin.messages' | translate }}</a></li>
-    <li><a>{{ 'admin.settings' | translate }}</a></li>
+    <li ui-sref-active="active"><a ui-sref="platformSettings">{{ 'admin.settings' | translate }}</a></li>
```
Comment (jredlarski) [inline src/admin/nav.html]: @mkwiatkowskisoldevelo Why did you add navigation to the settings page?

File: src/admin/bundles/bundle-settings.controller.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	'use strict';
+
+	angular.module('motech-admin')
+		.controller('BundleSettingsController', controller);
+
+	controller.$inject = ['$scope', '$stateParams', '$http', 'BundleSettingsFactory', 'BundleRawSettingsFactory', 'LoadingModal', 'ModalFactory', 'BundlesFactory'];
+	function controller($scope, $stateParams, $http, BundleSettingsFactory, BundleRawSettingsFactory, LoadingModal, ModalFactory, BundlesFactory){
+
+        var restartBundleHandler = function () {
+            $scope.module.$restart();
```
Comment (jredlarski) [inline src/admin/bundles/bundle-settings.controller.js]: Shouldn't it be `$scope.module.restart();`?
